### PR TITLE
bpo-34906: Doc: Fix typos (2)

### PR DESCRIPTION
**Pull request by matrixise**
Second turn for the typo in the doc, mainly for the Misc/ directory

https://bugs.python.org/issue34906

### CODE REVIEW

**Review by ned-deily — APPROVED** [review]
LGTM.  You'll probably need to make a manual backport to 3.7 and 3.6 of the subsets that apply to each.

File: Misc/HISTORY [hunk 1]
```diff
@@ -966,7 +966,7 @@ Core and Builtins
   returned NotImplemented.  Original patch by Martin Panter.
 
 - Issue #23321: Fixed a crash in str.decode() when error handler returned
-  replacment string longer than mailformed input data.
+  replacement string longer than mailformed input data.
```
Comment (vadmium) [inline Misc/HISTORY:969]: Bonus points for fixing malformed _mailformed_ spelling :)

**Review by terryjreedy — APPROVED** [review]
The idlelib corrections will backport cleanly.

File: Misc/HISTORY (hunk 1 above)
Comment (matrixise) [reply Misc/HISTORY:969]: oh yes, thank you. ;-)

**Comment by matrixise** [comment]
@vadmium here is my patch for `mailformed` thank you.

**Comment by matrixise** [comment]
For this PR and the previous one, I used `sphinxcontrib.spelling` and hunspell

**Comment by matrixise** [comment]
@ned-deily if you want, I can do the PR for 3.7 & 3.6, just keep me informed and I will work on it asap.

**Review by JulienPalard — APPROVED** [review]

**Comment by terryjreedy** [comment]
Ste'phane, are you planning to backport?  If not, I will do so for a couple of IDLE changes.

**Pull request merged**